src/tools/file-operations/smart-edit.ts

```
@@ -466,13 +404,37 @@ export class SmartEditTool {
+   * and one misses, applying the other two leaves the file in a state the
+   * caller never asked for and did not expect -- worse than doing nothing.
+   *
+   * Note this is only about PATTERN operations. A line-based edit whose content
+   * happens to equal what was already there is a genuine no-op and still
+   * returns success/unchanged.
    */
   private applyEdits(lines: string[], operations: EditOperation[]): string[] {
     // Sort operations by line number (descending) to avoid index shifting
     const sortedOps = [...operations].sort((a, b) => b.startLine - a.startLine);
 
     let result = [...lines];
```

ESLint

[error] 'result' is never reassigned. Use 'const' instead.

(prefer-const)